Marlin/src/HAL/AVR/fast_pwm.cpp

```diff
@@ -200,21 +208,15 @@ void set_pwm_frequency(const pin_t pin, int f_desired) {
         res = res_temp_fast;
         j = i;
         // Set the Wave Generation Mode to FAST PWM
-        if (timer.n == 2)
-          wgm = TERN(USE_OCR2A_AS_TOP, WGM2_FAST_PWM_OCR2A, WGM2_FAST_PWM);
-        else
-          wgm = WGM_FAST_PWM_ICRn;
+        wgm = (timer.n == 2) ? TERN(USE_OCR2A_AS_TOP, WGM2_FAST_PWM_OCR2A, WGM2_FAST_PWM) : WGM_FAST_PWM_ICRn;
```

Comment:
Cast compared values to `char` to avoid compile warning -Wenum-compare. 
```suggestion
        wgm = (timer.n == 2) ? (char)TERN(USE_OCR2A_AS_TOP, WGM2_FAST_PWM_OCR2A, WGM2_FAST_PWM) : (char)WGM_FAST_PWM_ICRn;
```